pyaedt/icepak.py

```diff
@@ -3772,3 +3776,240 @@ def assign_source(
+                    assignment_value = self._parse_variation_data(
+                        quantity,
+                        assignment["Type"],
+                        variation_value=assignment["Values"],
+                        function=assignment["Function"],
+                    )
+                    props.update(assignment_value)
+                else:
+                    if isinstance(assignment, (float, int)):
+                        assignment = str(assignment) + ["w_per_m2kel", "cel"][int(quantity == "Temperature")]
+                    props[quantity] = assignment
+        else:
+            props["Use External Conditions"] = False
+
+        if not boundary_name:
+            boundary_name = generate_unique_name(object_name + "_Block")
+
+        bound = BoundaryObject(self, boundary_name, props, "Block")
+        if bound.create():
+            self.boundaries.append(bound)
+            return bound
+
+    @pyaedt_function_handler
+    def assign_hollow_block(
+        self, object_name, assignment_type, assignment_value, boundary_name=None, external_temperature="AmbientTemp"
+    ):
+        """
+        Assign block boundary for hollow objects.
+
+        Parameters
+        ----------
+        object_name : str
+            Name of the object.
+        assignment_type : str
+            Type of the boundary assignment. The accepted types are: "Total Power", "Heat Flux",
+            "Temperature" or "Heat Transfer Coefficient".
+        assignment_value : str or dict
+            String with value and units of the assignment. If ``"Total Power"`` is assignment_type,
+            ``"Joule Heating"`` can be used.
+            Also, a dictionary can be used for temperature dependent or transient assignment.
+            The dictionary should contain three keys: ``"Type"``, ``"Function"`` and
+            ``"Values"``. Accepted ``"Type"`` values are: ``"Temp Dep"`` and ``"Transient"``.
+            Accepted ``"Function"`` are: ``"Linear"``, ``"Power Law"``, ``"Exponential"``,
+            ``"Sinusoidal"``, ``"Square Wave"`` and ``"Piecewise Linear"``. ``"Temp Dep"`` only
+            support the latter. ``"Values"`` contains a list of strings containing the parameters
+            required by the ``"Function"`` selection (e.g. ``"Linear"`` requires two parameters:
+            the value of the variable at t=0 and the slope of the line). The parameters required by
+            each ``Function`` option is in Icepak documentation. The parameters must contain the
+            units where needed.
+        boundary_name : str, optional
+            Name of the source boundary. The default is ``None`` and the boundary name will be
+            generated automatically.
+        external_temperature : str, dict or float, optional
+            String with value and unit of temperature for the Heat Transfer Coefficient. If float,
+            ``"cel"`` unit will be added automatically. The default is ``"AmbientTemp"``.
```

Comment:
```suggestion
        external_temperature : str, dict, or float, optional
            String with the value and unit of the temperature for the heat transfer coefficient.
            If a float value is specified, the ``"cel"`` unit is automatically added. The default
            is ``"AmbientTemp"``.
```
There is no description of the behavior if the data type is a dictionary?